client/state/data-layer/wpcom/sites/comments/likes/new/index.js

```diff
@@ -23,13 +25,29 @@ export const likeComment = ( { dispatch }, action ) => {
 	}, action ) );
 };
 
-export const updateCommentLikes = ( { dispatch }, { siteId, postId, commentId }, next, { like_count } ) => dispatch( local( {
-	type: COMMENTS_LIKE,
-	siteId,
-	postId,
-	commentId,
-	like_count
-} ) );
+export const updateCommentLikes = ( { dispatch }, { siteId, postId, commentId, status }, next, { like_count } ) => {
+	dispatch( local( {
+		type: COMMENTS_LIKE,
+		siteId,
+		postId,
+		commentId,
+		like_count
+	} ) );
+
+	if ( isEnabled( 'comments/undo-in-datalayer' ) && 'unapproved' === status ) {
+		const noticeId = `comment-notice-${ commentId }`;
+
+		dispatch( removeNotice( noticeId ) );
+
+		dispatch( successNotice( translate( 'Comment approved.' ), {
+			duration: 5000,
+			id: noticeId,
+			isPersistent: true,
+			button: translate( 'Undo' ),
+			onClick: () => dispatch( changeCommentStatus( siteId, postId, commentId, 'unapproved' ) )
```

Comment:
Because the like button is a toggle. If you want to undo a like, well, just click on it again. 🙂 

All the other actions, instead, move comments in other lists, making them disappear from the current view (mind that this does not happen with the CommentFaker).
So without an undo, if you accidentally approve a comment, you'd need to go to the approved list and hunt down the comment, which might not be in the first page.

At the same time, we don't really need a like/unlike notice. The behaviour is, in my opinion, well established everywhere. We might not use the thumbs-up or the heart icon, but its pretty clear what it does (of course I'll be disproved by the next user test on this :D ).

More importantly, like/unlike, by itself, is a "non-destructive" action (lack of better term). No matter its like status, the comment will remain in its place.
Though, this is not the case when liking unapproved comments. There, the comment is also moved to the approved list (which means: it shows up to all users).
As the approve action is "destructive", it has its own notice and undo. So we give to the "like+approve" action the same notice and undo (which of course should also undo the like; I guess my bad, somehow I thought unapproving also automatically unliked).